primitives/line.go

```diff
@@ -11,3 +13,38 @@ func (l Line) ToVector() Vector {
 func (l Line) Reverse() Line {
 	return Line{l.P2, l.P1}
 }
+
+func (l Line) IsCollinearPointOnSegment(p Point) bool {
+	// Works only for collinear points
+	if (p.X <= math.Max(l.P1.X, l.P2.X) && p.X >= math.Max(l.P1.X, l.P2.X)) && 
+	   (p.Y <= math.Max(l.P1.Y, l.P2.Y) && p.Y >= math.Max(l.P1.Y, l.P2.Y)) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (l Line) IsIntersectingSegment(l1 Line) bool {
+	// https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+
+	orientation1 := l1.P1.Orientation(l.P1, l.P2)
+	orientation2 := l1.P2.Orientation(l.P1, l.P2)
```

Comment:
check please Leyla's PR for IntersectLine function. Let's leave one that is better.